utils/src/main/kotlin/Extensions.kt

```diff
@@ -370,6 +370,20 @@ fun String.stripCredentialsFromUrl() =
         this
     }
 
+/**
+ * If this string starts with [prefix], return success and the string without the prefix, otherwise return a failure.
+ */
+fun String?.withoutPrefix(prefix: String): Result<String> {
```

Comment:
Not sure whether I have grasped all the potential use cases of this function, but I was wondering whether it was not easier to simply return a nullable String rather than a `Result<String>`. This would simplify the evaluation of the function result even more; at least in the cases where the function is used so far.

My impression is that only two cases are relevant: the passed in String starts with the prefix or not (a *null* input string would be just a special case for the second category). This binary logic would be expressed in a more straight-forward manner with a nullable result.

Reply:
Good point, this probably was a bit over-engineered. I've simplified the implementation.